### DISCUSSION TITLE
Residual parameters 'saem'

### DISCUSSION THREAD

**Discussion by simonbeyer1** [nlmixr2/nlmixr2#245] [General]
Dear nlmixr community,

Why is it possible to fit residual parameters on log-scale when using 'focei' but not with 'saem'? Is this a feature or a bug? 

Small example: 
```
## The basic model consists of an ini block that has initial estimates
library(nlmixr2)
setwd(tempdir())
one.compartment <- function() {
  ini({
    tka <- log(2); label("Ka")
    tcl <- log(2.72); label("Cl")
    tv <- log(31.5); label("V")
    
    eta.ka ~ 0.6
    eta.cl ~ 0.3
    eta.v ~ 0.1
    log.rel.sd <- log(0.1)
  })
  # and a model block with the error specification and model specification
  model({
    ka <- exp(tka + eta.ka)
    cl <- exp(tcl + eta.cl)
    v <- exp(tv + eta.v)
    rel.sd <- exp(log.rel.sd)
    d/dt(depot) <- -ka * depot
    d/dt(center) <- ka * depot - cl / v * center
    cp <- center / v
    cp ~ prop(rel.sd)
  })
}

fit.focei <- nlmixr2(one.compartment, theo_sd, est="focei", control = foceiControl(print = 0))
fit.saem <- nlmixr2(one.compartment, theo_sd, est="saem", control = saemControl(print = 0), table = tableControl(cwres = T))
```
The focei fit is working fine. The 'saem' call gives the error message
```
Error : 'one.compartment' residual parameters cannot depend on the model calculated parameters for the estimation routine 'saem'
```
Thanks in advance :)
Best,
Simon

**Comment by mattfidler** [comment]
Hi @simonbeyer1 

This is a missing feature from `saem`.  `saem` handles residuals quite differently than `focei` and therefore at the current time this is not supported.

**Reply by mattfidler** [reply]
As to why, `saem` does not estimate residuals for many cases but calculates them based on the MCMC walk.  The exceptions to this are `add` + `prop` models or `boxCox()` / `yeoJohnson()` transformations where the residuals are optimized with nealder mead downhill simplex.  Getting a "better" estimate with log-transformed values with `saem` just doesn't work.
